src/transport/SecureSession.h

```diff
@@ -194,7 +195,8 @@ class SecureSession : public Session, public ReferenceCounted<SecureSession, Sec
             const auto & defaultMRRPConfig   = GetDefaultMRPConfig();
             const auto & localMRPConfig      = maybeLocalMRPConfig.ValueOr(defaultMRRPConfig);
             return GetRetransmissionTimeout(localMRPConfig.mActiveRetransTimeout, localMRPConfig.mIdleRetransTimeout,
-                                            ourLastActivity, localMRPConfig.mActiveThresholdTime);
+                                            ourLastActivity, localMRPConfig.mActiveThresholdTime,
+                                            ourLastActivity == System::Clock::kZero /*isFirstMessageOnExchange*/);
```

Comment:
```suggestion
                                            false /*isFirstMessageOnExchange*/);
```
`GetMessageReceiptTimeout` is always asking how long the _response_ will take, so it's never the first message on an exchange, no?